error: ‘memcpy’ was not declared in this scope

Hi!

I got several compilation problems like this:

```
In function ‘plusaes::detail::RoundKeys plusaes::detail::expand_key(const unsigned char*, int)’:
.../plusaes/include/plusaes/plusaes.hpp:236:9: error: ‘memcpy’ was not declared in this scope
  236 |         memcpy(&w[i], key + (i * kWordSize), kWordSize);
      |         ^~~~~~
```

I managed to compile by including `cstring` in `plusaes.hpp` as follows:

```
#include <cstring>
``` 

My compiler/OS are:

```
$ gcc --version
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
```

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,2 +1,5 @@
 # Auto detect text files and perform LF normalization
 * text=auto
+
+# Consider building on Docker on Windows.
+lin/**/* text eol=lf

--- a/include/plusaes/plusaes.hpp
+++ b/include/plusaes/plusaes.hpp
@@ -7,6 +7,7 @@
 #ifndef PLUSAES_HPP__
 #define PLUSAES_HPP__
 
+#include <cstring>
 #include <stdexcept>
 #include <vector>
 

--- a/lin/build.sh
+++ b/lin/build.sh
@@ -1,0 +1,5 @@
+#!/bin/bash
+
+./bootstrap
+./configure
+make

--- a/lin/unit_test/Makefile.am
+++ b/lin/unit_test/Makefile.am
@@ -18,5 +18,5 @@ unit_test_SOURCES = \
   $(SRCDIR)/gtest/gtest-all.cc \
   $(SRCDIR)/main.cpp \
   $(SRCDIR)/test-ctr.cpp \
-  $(SRCDIR)/test-plusaes.cpp
-
+  $(SRCDIR)/test-plusaes.cpp \
+  $(SRCDIR)/test-inc.cpp

--- a/mac/plusaes.xcodeproj/project.pbxproj
+++ b/mac/plusaes.xcodeproj/project.pbxproj
@@ -16,6 +16,7 @@
 		9E68E6C124A776A600F953F9 /* test-plusaes.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 9E6747BA2439B4690007285B /* test-plusaes.cpp */; };
 		9E68E6C224A776A600F953F9 /* test-ctr.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 9E446E7E2442A2C5000D333A /* test-ctr.cpp */; };
 		9E68E6C324A7771A00F953F9 /* plusaes.hpp in Sources */ = {isa = PBXBuildFile; fileRef = 004FC8FA1BC5E37C005D4DD3 /* plusaes.hpp */; };
+		9EF1C820260ECF9C0056CA87 /* test-inc.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 9EF1C81F260ECF9C0056CA87 /* test-inc.cpp */; };
 /* End PBXBuildFile section */
 
 /* Begin PBXCopyFilesBuildPhase section */
@@ -41,6 +42,7 @@
 		9E68E6B824A774E600F953F9 /* unit_test_target.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = unit_test_target.xctest; sourceTree = BUILT_PRODUCTS_DIR; };
 		9E68E6BA24A774E600F953F9 /* unit_test_target.m */ = {isa = PBXFileReference; explicitFileType = sourcecode.cpp.objcpp; path = unit_test_target.m; sourceTree = "<group>"; };
 		9E68E6BC24A774E600F953F9 /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
+		9EF1C81F260ECF9C0056CA87 /* test-inc.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = "test-inc.cpp"; sourceTree = "<group>"; };
 /* End PBXFileReference section */
 
 /* Begin PBXFrameworksBuildPhase section */
@@ -135,8 +137,9 @@
 			children = (
 				9E6747AE2439B4690007285B /* gtest */,
 				9E6747BB2439B4690007285B /* main.cpp */,
-				9E6747BA2439B4690007285B /* test-plusaes.cpp */,
 				9E446E7E2442A2C5000D333A /* test-ctr.cpp */,
+				9EF1C81F260ECF9C0056CA87 /* test-inc.cpp */,
+				9E6747BA2439B4690007285B /* test-plusaes.cpp */,
 			);
 			path = src;
 			sourceTree = "<group>";
@@ -249,6 +252,7 @@
 			files = (
 				9E6747C62439B4690007285B /* main.cpp in Sources */,
 				9E6747BC2439B4690007285B /* gtest-all.cc in Sources */,
+				9EF1C820260ECF9C0056CA87 /* test-inc.cpp in Sources */,
 				9E6747C52439B4690007285B /* test-plusaes.cpp in Sources */,
 				9E446E7F2442A2C5000D333A /* test-ctr.cpp in Sources */,
 			);

--- a/unit_test/src/test-inc.cpp
+++ b/unit_test/src/test-inc.cpp
@@ -1,0 +1,1 @@
+#include "plusaes/plusaes.hpp"

--- a/win/unit_test/unit_test.vcxproj
+++ b/win/unit_test/unit_test.vcxproj
@@ -169,6 +169,7 @@
     <ClCompile Include="..\..\unit_test\src\gtest\gtest-all.cc" />
     <ClCompile Include="..\..\unit_test\src\main.cpp" />
     <ClCompile Include="..\..\unit_test\src\test-ctr.cpp" />
+    <ClCompile Include="..\..\unit_test\src\test-inc.cpp" />
     <ClCompile Include="..\..\unit_test\src\test-plusaes.cpp" />
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />

--- a/win/unit_test/unit_test.vcxproj.filters
+++ b/win/unit_test/unit_test.vcxproj.filters
@@ -41,5 +41,8 @@
     <ClCompile Include="..\..\unit_test\src\test-ctr.cpp">
       <Filter>src</Filter>
     </ClCompile>
+    <ClCompile Include="..\..\unit_test\src\test-inc.cpp">
+      <Filter>src</Filter>
+    </ClCompile>
   </ItemGroup>
 </Project>

